### PR TITLE
SSO docs cleanups

### PR DESCRIPTION
**Pull request by toolmantim**
This reworks the SSO overview page to make it a bit more of a friendly overview, include more screenshots, and more clearly link to all the provider-specific guides (GitHub is the only one missing now).

Before: https://buildkite.com/docs/integrations/sso

After: https://buildkite-docs-review-pr-506.herokuapp.com/docs/integrations/sso

### CODE REVIEW

**Comment by toolmantim** [comment]
Could I get your help to get this one passing and 🚀’d @harrietgrace & @eleanorakh?

A few things I noticed:

* The lack of a GitHub page really stands out now
* Each individual provider doc doesn’t state what it’s support is for admin/team attributes
* We could add all the providers to the self-serve UI index list, even if they have to be set up manually and/or via support. The “new” page could link to the doc with manual setup instructions, or to mention contacting support.

**Comment by eleanorakh** [comment]
@toolmantim This looks ace! A couple of questions:

>The lack of a GitHub page really stands out now.

I'm not sure what you mean by this, GitHub is a supported provider for self-serve SSO?

>Each individual provider doc doesn’t state what it’s support is for admin/team attributes

There's a section in the FAQ: "Can I sync my identity provider's groups with my Buildkite teams?
Yes, if you're using a SAML identity provider. OAuth providers such as Google G Suite do not support groups"

Do you think it's necessary to add something like this to each provider page?

>We could add all the providers to the self-serve UI index list, even if they have to be set up manually and/or via support. The “new” page could link to the doc with manual setup instructions, or to mention contacting support

The only one that's currently listed on the existing single-sign on doc is Google Cloud Identity. It's a SAML setup, so there's a couple of options: link the new page to the custom SAML settings, or link it to the [docs](https://buildkite-docs-review-pr-506.herokuapp.com/docs/integrations/sso/g-cloud-identity)..

**Comment by toolmantim** [comment]
Great q's!

>> The lack of a GitHub page really stands out now.
> I'm not sure what you mean by this, GitHub is a supported provider for self-serve SSO?

Ah, I was referring to it standing out here in the docs, as missing a page on the new overview:

image

and here in the docs nav:

image

For context, a GitHub page in the SSO section was part of the initial PR back before self-serve SSO, but we didn't want to encourage it because we thought it'd be confusing that we have two ways to connect to GitHub: the new GitHub App for login (which eventually handle everything), and the old GitHub App we use for commit statuses currently and which you connect via "Connected Apps" in your personal settings.

I think the cat is out of the bag now that GitHub is part of the self-serve UI in the app, so we should probably have a docs page for it? And it can help to explain why there's two separate GitHub apps perhaps (one for login, one for commit status updating).

>> Each individual provider doc doesn’t state what it’s support is for admin/team attributes
> There's a section in the FAQ: "Can I sync my identity provider's groups with my Buildkite teams?
Yes, if you're using a SAML identity provider. OAuth providers such as Google G Suite do not support groups"

Yeah, that's the part that I thought was confusing. I was just thinking that if each provider had different features it supported "Team sync: yes" type thing, it'd make sense per-provider.

For now, in this PR, I changed the FAQ to at least just make it sound like it's a "Custom SAML Provider" only thing (vs just any SAML-based provider):

    ### Can I sync my identity provider's groups with my Buildkite teams?
    
    Yes, if you're using the Custom SAML provider you can configure SAML User Attributes.

> The only one that's currently listed on the existing single-sign on doc is Google Cloud Identity. It's a SAML setup, so there's a couple of options: link the new page to the custom SAML settings, or link it to the docs.

Oh nice, it's only missing one!

I just noticed, but I think the Google Cloud Identity docs are wrong too, because it says you have to email support with all the certificate and other details, but I think you're able to use the Custom SAML provider in the UI yeah?
https://buildkite-docs-review-pr-506.herokuapp.com/docs/integrations/sso/g-cloud-identity

**Comment by harrietgrace** [comment]
I think I've got a Github page floating around somewhere. It wouldn't take long to put one together if you wanted that for launch - it's pretty much the same as the Google G Suite page. 

We could add teams stuff to the Okta page, but that's about it. I'm not across the specifics of how to do it for any of the other providers. 

I'm not sure how I feel about making that FAQ only about custom SAML... Maybe we could leave it as an any-SAML-provider thing but make it more "if you know how to send through additional attributes in your SAML assertion then you can totes do teams stuff". 

The tl;dr of adding teams is that they need to associate their team's UUID from Buildkite with their IdP's group, then send the UUID in their SAML assertion with each login. So it's not overly complex, but the person needs to know what they're about with their IdP settings. 

I left the G Cloud Identity docs as they were because I assumed that at some point we'd add it as its own self serve button. Totes happy to update it to point through custom SAML though.

**Comment by eleanorakh** [comment]
>The lack of a GitHub page really stands out now.

Ahh, I see now! Yep, having a dedicated GitHub page would be appropriate.

> But the FAQ is probably okay for now? 🤷🏼‍♂️

Yep! I think that covers it for the time being :)

>I just noticed, but I think the Google Cloud Identity docs are wrong too, because it says you have to email support with all the certificate and other details, but I think you're able to use the Custom SAML provider in the UI yeah?

Correct, that can be done with custom SAML. Maybe we can remove the 'Email Buildkite' section and replace it with section 2 from [here](https://buildkite-docs-review-pr-506.herokuapp.com/docs/integrations/sso/custom-saml)

**Comment by toolmantim** [comment]
That all makes sense! We can do GitHub as a follow up, and just adjust this as you see fit @harrietgrace.

**Comment by harrietgrace** [comment]
@toolmantim @eleanorakh  I've updated the Cloud Identity doc to direct people through the Custom SAML self-serve stuff. They'll still need to email us to get their SSO url bizzo, the same as with ADFS and other custom providers. 

I've tried out some new words for that FAQ about teams. It turned out a bit technical and wordy, but I wanted it to sound like it's a bit involved (because it is). What do y'all think?

**Comment by eleanorakh** [comment]
@harrietgrace I'm really happy with those changes. It's a good call to make the teams FAQ question more technical and involved.

**Comment by toolmantim** [comment]
Thanks @harrietgrace & @eleanorakh!
